config/initializers/devise.rb

```diff
@@ -7,6 +7,7 @@
   # Devise will use the `secret_key_base` as its `secret_key`
   # by default. You can change it below and use your own secret key.
   # config.secret_key = '....'
+  config.secret_key = "ENV[DEVISE_SECRET_KEY]"
```

Comment:
It should get this from config.secret_key_base. This key isn't set on production so it will fail there.